### PR TITLE
[TS SDK] Update transferWithOptIn function to use remote abi

### LINKED ISSUE #7075: [TS SDK] have `token_client` to use remote ABI
TS SDK now uses remote ABI to interact with entry functions in a smart contract. `token_client` still uses local ABI to use `transfer_with_opt_in` entry function 
https://github.com/aptos-labs/aptos-core/blame/main/aptos-move/framework/aptos-token/sources/token.move#L485-L498

Change it to use the remote ABI instead

### PR DESCRIPTION
**Pull request by jin-aptos**
### Description

1. Update transferWithOptIn function to use remote abi - issue Close #7075 
2. Fixed the unit test to prevent `ApiError: Too Many Requests` error when testing again local testnet

#### pnpm test
Test Suites: 20 passed, 20 total
Tests:       174 passed, 174 total
Snapshots:   0 total
Time:        152.805 s

### CODE REVIEW

File: ecosystem/typescript/sdk/src/plugins/token_client.ts [hunk 1]
```diff
@@ -396,23 +388,15 @@ export class TokenClient {
...
-          new TransactionArgumentU8Vector(new TextEncoder().encode(tokenName)),
-          new TransactionArgumentU64(BigInt(propertyVersion)),
-          new TransactionArgumentAddress(TxnBuilderTypes.AccountAddress.fromHex(receiver)),
-          new TransactionArgumentU64(BigInt(amount)),
-        ],
-      ),
+    const builder = new TransactionBuilderRemoteABI(this.aptosClient, { sender: sender.address(), ...extraArgs });
+    const rawTxn = await builder.build(
+      "0x3::token::transfer_with_opt_in", 
+      [],
```
Comment (jin-aptos) [inline ecosystem/typescript/sdk/src/plugins/token_client.ts:394]: I assume there is no _ty_tags_ needed here. But I am curious when is this needed?

File: ecosystem/typescript/sdk/src/plugins/token_client.ts [hunk 2]
```diff
@@ -10,16 +10,8 @@ import * as Gen from "../generated/index";
 import { HexString, MaybeHexString } from "../utils";
 import { TransactionBuilder, TransactionBuilderRemoteABI, TxnBuilderTypes } from "../transaction_builder";
 import { MAX_U64_BIG_INT } from "../bcs/consts";
-import { TOKEN_TRANSFER_OPT_IN } from "../abis";
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk/src/plugins/token_client.ts:13]: I think this is the only const in this file, can we delete the file...?

File: ecosystem/typescript/sdk/src/plugins/token_client.ts (hunk 1 above)
Comment (0xmaayan) [reply ecosystem/typescript/sdk/src/plugins/token_client.ts:394]: `type_arguments` needed when the move function expects typed arguments, i.e https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token/sources/token_coin_swap.move#L103

if it doesn't we simply pass in an empty array - though we should probably be more sophisticate than that as with the remote abi we should know if the function expects it, and if it doesn't we should be able to not even ask for it, but that is a different discussion not related to this PR

**Review by 0xmaayan — APPROVED** [review]
AMAZING!

File: ecosystem/typescript/sdk/src/plugins/token_client.ts (hunk 2 above)
Comment (jin-aptos) [reply ecosystem/typescript/sdk/src/plugins/token_client.ts:13]: You mean removing the `abis.ts` file? Yes!

**Comment by jin-aptos** [comment]
Also fixed the unit test to prevent `ApiError: Too Many Requests` error when testing again local testnet

File: ecosystem/typescript/sdk/src/tests/e2e/indexer.test.ts [hunk 3]
```diff
@@ -4,15 +4,15 @@ import { bcsSerializeBool } from "../../bcs";
 import { FaucetClient } from "../../plugins/faucet_client";
 import { IndexerClient } from "../../providers/indexer";
 import { TokenClient } from "../../plugins/token_client";
-import { API_TOKEN, longTestTimeout } from "../unit/test_helper.test";
+import { FAUCET_AUTH_TOKEN, longTestTimeout } from "../unit/test_helper.test";
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk/src/tests/e2e/indexer.test.ts:7]: thanks for fixing it! maybe worth getting it to test_helper file, something like `getTestnetFaucet()` https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/tests/unit/test_helper.test.ts

Comment (0xmaayan) [reply ecosystem/typescript/sdk/src/tests/e2e/indexer.test.ts:7]: yes, a new `getTestnetFaucet()`. So all tests run against local node, but indexer tests runs against testnet (this is b/c spinning up a local indexer is almost a headache).  `getFaucetClient()` uses local node endpoints and we want to keep it like that. 

BUT feel free to leave it for a different/second PR

**Pull request merged**